client/src/main/java/org/apache/rocketmq/client/impl/consumer/AssignedMessageQueue.java

```diff
@@ -83,10 +83,15 @@ public long getPullOffset(MessageQueue messageQueue) {
         return -1;
     }
 
-    public void updatePullOffset(MessageQueue messageQueue, long offset) {
+    public void updatePullOffset(MessageQueue messageQueue, long offset, ProcessQueue processQueue) {
         MessageQueueState messageQueueState = assignedMessageQueueState.get(messageQueue);
-        if (messageQueueState != null) {
-            messageQueueState.setPullOffset(offset);
+        synchronized (this.assignedMessageQueueState) {
+            if (messageQueueState.getProcessQueue() != processQueue) {
+                return;
+            }
+            if (messageQueueState != null) {
+                messageQueueState.setPullOffset(offset);
+            }
         }
```

Comment:
Synchronized here seems to be unnecessary

Reply:
it seems putting the synchronized before assignedMessageQueueState.get will be better.

Reply:
It seems the synchronized lock can be removed. 
I have update this branch.